django/contrib/auth/base_user.py

```diff
@@ -60,9 +61,20 @@ def get_username(self):
         "Return the identifying username for this User"
         return getattr(self, self.USERNAME_FIELD)
 
+    def __init__(self, *args, **kwargs):
+        super(AbstractBaseUser, self).__init__(*args, **kwargs)
+        # Stores the raw password if set_password() is called so that it can
+        # be passed to password_changed() after the model is saved.
+        self._password = None
+
     def __str__(self):
         return self.get_username()
 
+    def save(self, *args, **kwargs):
+        super(AbstractBaseUser, self).save(*args, **kwargs)
+        if self._password is not None:
+            password_validation.password_changed(self._password, self)
+
```

Comment:
I think we should clear `self._password` here to a) minimize the lifespan of the raw password in memory and b) avoid double-calling `password_changed` in case of multiple subsequent saves of the same user instance.